python_modules/dagster-graphql/dagster_graphql/schema/asset_graph.py

```diff
@@ -528,28 +529,116 @@ def is_graph_backed_asset(self) -> bool:
     def is_executable(self) -> bool:
         return self._asset_node_snap.is_executable
 
-    def get_materialization_status_for_asset_health(self, graphene_info: ResolveInfo):
-        # if non-partitioned:
-        # if latest materialization succeed, healthy
-        # if latest materialization failed, degraded
-        # no materialization, unknown
-        # if partitioned:
-        # if all partitions successfully materialized, healthy
-        # if any partitions missing, warning
-        # if any partitions failed, degraded
-        if (
-            not graphene_info.context.instance.dagster_observe_supported()
-            or not graphene_info.context.instance.can_read_failure_events_for_asset(
-                self._asset_node_snap.asset_key
-            )
+    async def get_materialization_status_for_asset_health(self, graphene_info: ResolveInfo):
+        if self._asset_node_snap.partitions is not None:  # isPartitioned
+            # TODO - maybe can do this from the asset record? that might not re-gen the cache though
```

Comment:
confirming that that does not re-gen the cache - the record is used as an input to regen the cache